code/game/machinery/floor_light.dm

```diff
@@ -171,3 +173,44 @@ var/list/floor_light_cache = list()
 /obj/machinery/floor_light/cultify()
 	default_light_colour = "#FF0000"
 	update_brightness()
+
+/obj/machinery/floor_light/changing
+	name = "changing Floor light"
+	var/last_color_change
+
+/obj/machinery/floor_light/changing/Initialize(mapload)
+	. = ..()
+	last_color_change = world.time
+
+/obj/machinery/floor_light/changing/process(delta_time)
+	. = ..()
+	update_colour()
+
+/obj/machinery/floor_light/changing/proc/update_colour()
+	if(last_color_change > world.time - 0.5 SECONDS)
```

Comment:
Process only runs every two seconds on machinery, no?

Reply:
Dunno, but either way enough
It doesn't exactly matter how fast it changes, just often enough to be noticable